pkgs/by-name/fr/freedownloadmanager/package.nix

```diff
@@ -0,0 +1,124 @@
+# pkgs/by-name/fr/freedownloadmanager/package.nix
+{
+  lib,
+  stdenv,
+  fetchurl,
+  dpkg,
+  wrapGAppsHook3,
+  autoPatchelfHook,
+  udev, # Needed for lib.getLib below? wrapGAppsHook might suffice
+  libdrm,
+  libpqxx,
+  unixODBC,
+  gst_all_1,
+  libpulseaudio,
+  xcbutilcursor,
+  xcbutilwm,
+  xcbutilimage,
+  xcbutilkeysyms,
+  xcbutilrenderutil,
+  patchelf, # needed for manual workaround
+  copyDesktopItems, # standard hook for .desktop files
+  makeWrapper, # standard hook for wrapping executables
+}:
+
+stdenv.mkDerivation rec {
+  pname = "freedownloadmanager";
+  # Version number corresponds to the hash below as of 2025-04-05.
+  # Upstream uses `/latest/` URL which isn't ideal for reproducibility.
+  version = "6.19.0";
+
+  # NOTE: Fetches from a `/latest/` URL. If upstream updates the file at
+  # this URL *for this version*, the hash will need updating.
+  # A stable, versioned URL (e.g., .../6.19.0/freedownloadmanager.deb)
+  # would be preferable if found upstream.
+  src = fetchurl {
+    url = "https://files2.freedownloadmanager.org/6/latest/freedownloadmanager.deb";
+    hash = "sha256-yjyRB/4vnVXB8ZmicZxsrENCCUtp7bRR0YY5x0JtmTg=";
+  };
+
+  unpackPhase = "dpkg-deb -x $src .";
+
+  nativeBuildInputs = [
+    dpkg
+    autoPatchelfHook
+    patchelf
+    wrapGAppsHook3
+    copyDesktopItems
+    makeWrapper
+  ];
+
+  buildInputs =
+    [
+      # Runtime libraries identified as needed by FDM or its bundled Qt libs
+      libdrm
+      libpqxx # Bundled Qt SQL plugin dependency (PostgreSQL)
+      unixODBC # Bundled Qt SQL plugin dependency (ODBC)
+      libpulseaudio # For QtMultimedia
+      # XCB utils needed by Qt platform plugin
+      xcbutilcursor
+      xcbutilwm
+      xcbutilimage
+      xcbutilkeysyms
+      xcbutilrenderutil
+      (lib.getLib udev) # For device access, if needed beyond what hooks provide
+    ]
+    ++ (with gst_all_1; [
+      gstreamer
+      gst-libav
+      gst-plugins-base
+      gst-plugins-good
+      gst-plugins-bad
+      gst-plugins-ugly
+    ]);
+
+  # Note: autoPatchelfIgnoreMissing = [ ... ] failed during testing (NixOS unstable 24.11, 2025-04-05).
+  # So I used a manual patchelf in installPhase as a workaround. You can see my commit message for more information.
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/opt/${pname} $out/share/pixmaps
+
+    cp -r opt/freedownloadmanager/* $out/opt/${pname}/
+    cp -r usr/share/* $out/share/
+
+    ln -s $out/opt/${pname}/fdm $out/bin/${pname}
+
+    ln -s $out/opt/${pname}/icon.png $out/share/pixmaps/${pname}.png
+
+    substituteInPlace $out/share/applications/freedownloadmanager.desktop \
```

Comment:
It might be better to use [desktop-file-utils](https://www.freedesktop.org/wiki/Software/desktop-file-utils/) here, for an example see [gitbutler/package.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/gi/gitbutler/package.nix#L150).

Reply:
(or [makeDesktopItem](https://github.com/NixOS/nixpkgs/blob/master/doc/build-helpers/trivial-build-helpers.chapter.md#usage-2-of-makedesktopitem) but that might be a bit overkill here)